slick-codegen/src/main/scala/slick/codegen/AbstractGenerator.scala

```diff
@@ -149,7 +149,7 @@ abstract class AbstractGenerator[Code,TermName,TypeName](model: m.Model)
       /** The * projection that accumulates all columns and map them if mappingEnabled is true*/
       def star: Code
       /** Indicates whether a ? projection should be generated. */
-      def optionEnabled = mappingEnabled && columns.exists(c => !c.model.nullable)
+      def optionEnabled = mappingEnabled && columns.exists(c => !c.model.nullable) && (columns.size <= 22 || hlistEnabled)
```

Comment:
this is better than changing `def option` to generate empty string. Because otherwise the docs for the def will still be generated.